waku/common/postgres/connection.nim

```diff
@@ -0,0 +1,106 @@
+when (NimMajor, NimMinor) < (1, 4):
+  {.push raises: [Defect].}
+else:
+  {.push raises: [].}
+
+import
+  stew/results,
+  chronicles,
+  chronos
+
+import ./common
+
+include db_postgres
+
+
+logScope:
+  topics = "postgres connection"
+
+
+## Connection options
+
+type PgConnOptions* = object
+    connection: string
+    user: string
+    password: string
+    database: string
+
+func init*(T: type PgConnOptions, connection, user, password, database: string): T =
+  PgConnOptions(
+    connection: connection,
+    user: user,
+    password: password,
+    database: database
+  )
+
+func connection*(options: PgConnOptions): string =
+  options.connection
+
+func user*(options: PgConnOptions): string =
+  options.user
+
+func password*(options: PgConnOptions): string =
+  options.password
+
+func database*(options: PgConnOptions): string =
+  options.database
+
+
+## Connection management
+
+proc error(db: DbConn): string =
+  ## Extract the error message from the database connection.
+  $pqErrorMessage(db)
+
+
+proc open*(options: PgConnOptions): common.PgResult[DbConn] =
```

Comment:
This proc is not actually async, right?

Reply:
No... It is _sync_. Just a convenience wrapper that takes the Pg options and returns a result.

Reply:
So personally I would just have a `numConnections` instead of `min/max`, don't want my program to randomly do blocking IO when I reach an arbitrary limit. Since the connections >min are not closed, you would probably end up with always `maxConnections` anyway
Though would also have to check what happens if for instance, the server restarts

Reply:
> Though would also have to check what happens if, for instance, the server restarts (or worst, just shuts down)

Absolutely! There are a thousand corner cases that I am not covering here. But for time's sake, I am trying to have a little MVP covering the "happy-path" cases, and we can iteratively improve the implementation.